runtime/jvmti/jvmtiMemory.c

```diff
@@ -87,19 +87,15 @@ jvmtiSetHeapSamplingInterval(jvmtiEnv *env,
 {
 	jvmtiError rc = JVMTI_ERROR_NONE;
 	J9VMThread *currentThread = NULL;
-	J9JavaVM *vm = JAVAVM_FROM_ENV(env);
 	
 	Trc_JVMTI_jvmtiSetHeapSamplingInterval_Entry(env, samplingInterval);
 	
 	ENSURE_PHASE_ONLOAD_OR_LIVE(env);
 	ENSURE_CAPABILITY(env, can_generate_sampled_object_alloc_events);
 	ENSURE_NON_NEGATIVE(samplingInterval);
 
-	rc = getCurrentVMThread(vm, &currentThread);
-	if ((JVMTI_ERROR_NONE == rc) && (NULL != currentThread)) {
-		/* No negative samplingInterval, and there is no data lost when jint is casted to UDATA. */
-		vm->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(currentThread, samplingInterval);
-	}
+	/* No negative samplingInterval, and there is no data lost when jint is casted to UDATA. */
+	currentThread->javaVM->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(((J9JVMTIEnv *)env)->vm, samplingInterval);
```

Comment:
VM is fetched twice here, in two different ways. Please use a local.

Reply:
Fixed, please have another look.